### ISSUE TITLE
custom backward function with upstream gradients.

### ISSUE THREAD

**Issue by ByungSun12** [PyTorchLightning/pytorch-lightning#3695]
#### What is your question?
How to implement `backward` with upstream gradients?

I want to implement `multi-staged backpropagation` with pytorch-lightning.

This is my implementation code with pytorch.

[My implementation](https://gist.github.com/ByungSun12/ad964a08eba6a7d103dab8588c9a3774#file-cifar10_tutorial_with_multistaged_backpropagation-py)

And I want to convert this code (that is below) to pytorch-lightning version.

As following `multi-staged backpropagation`, I need to implement explicit chain rule.

But I don't know how to write code in pytorch-lightning version.

#### Code
(AS-IS)
   
![스크린샷 2020-09-28 오후 1 53 46](https://user-images.githubusercontent.com/65884594/94392001-0b37ee80-0192-11eb-9887-0690a09cc92b.png)

(TO-DO)
```{python}
def training_loop(self, batch, batch_idx):
    # I want to write code for multi-staged backpropagation...
    pass
```

**Comment by ByungSun12** [comment]
Sorry.. Closed Issue. I solve this. I just write code `backward` method overriding...
